cluster/src/main/java/com/hartwig/pipeline/tertiary/cider/Cider.java

```diff
@@ -0,0 +1,215 @@
+import com.hartwig.pipeline.output.Folder;
+import com.hartwig.pipeline.output.RunLogComponent;
+import com.hartwig.pipeline.reruns.PersistedDataset;
+import com.hartwig.pipeline.reruns.PersistedLocations;
+import com.hartwig.pipeline.resource.ResourceFiles;
+import com.hartwig.pipeline.stages.Namespace;
+import com.hartwig.pipeline.storage.GoogleStorageLocation;
+import com.hartwig.pipeline.storage.RuntimeBucket;
+import com.hartwig.pipeline.tertiary.TertiaryStage;
+import com.hartwig.pipeline.tools.HmfTool;
+
+@Namespace(Cider.NAMESPACE)
+public class Cider extends TertiaryStage<CiderOutput> {
+
+    public static final String NAMESPACE = "cider";
+
+    private final ResourceFiles resourceFiles;
+    private final PersistedDataset persistedDataset;
+
+    public Cider(final AlignmentPair alignmentPair, final ResourceFiles resourceFiles, final PersistedDataset persistedDataset) {
+        super(alignmentPair);
+        this.resourceFiles = resourceFiles;
+        this.persistedDataset = persistedDataset;
+    }
+
+    @Override
+    public String namespace() {
+        return NAMESPACE;
+    }
+
+    @Override
+    public List<BashCommand> tumorReferenceCommands(final SomaticRunMetadata metadata) {
+        return tumorOnlyCommands(metadata);
+    }
+
+    @Override
+    public List<BashCommand> tumorOnlyCommands(final SomaticRunMetadata metadata) {
+        List<String> arguments = ciderArguments(metadata);
+        // only need 8G for DNA samples. RNA samples would need 32G
+        return List.of(new JavaJarCommand(HmfTool.CIDER, arguments));
+    }
+
+    private List<String> ciderArguments(final SomaticRunMetadata metadata) {
+        List<String> arguments = new ArrayList<>();
+        arguments.add(String.format("-sample %s", metadata.tumor().sampleName()));
+        arguments.add(String.format("-bam %s", getTumorBamDownload().getLocalTargetPath()));
+        arguments.add(String.format("-ref_genome %s", resourceFiles.refGenomeFile()));
+        arguments.add(String.format("-ref_genome_version %s", resourceFiles.version()));
+        arguments.add(String.format("-blast %s", BLASTN.path()));
+        arguments.add(String.format("-blast_db %s", resourceFiles.blastDb()));
+        arguments.add("-write_cider_bam");
+        arguments.add(String.format("-output_dir %s", VmDirectories.OUTPUT));
+        arguments.add(String.format("-threads %s", Bash.allCpus()));
+        return arguments;
+    }
+
+    @Override
+    public VirtualMachineJobDefinition vmDefinition(final BashStartupScript startupScript, final ResultsDirectory resultsDirectory) {
+        return ImmutableVirtualMachineJobDefinition.builder()
+                .name("cider")
+                .startupCommand(startupScript)
+                .performanceProfile(custom(HmfTool.CIDER.getCpus(), HmfTool.CIDER.getMemoryGb()))
+                .namespacedResults(resultsDirectory)
+                .build();
+    }
+
+    // <sampleId>.cider.layout.gz
```

Comment:
unclear what these comments mean, generally we don't leave comments in this code.